test/foundry/deploy/15_holograph_registry_tests.t.sol

```diff
@@ -0,0 +1,802 @@
+
+        bool found = false;
+        for (uint i = 0; i < contracts.length; i++) {
+            if (contracts[i] == address(sampleErc721Holographer)) {
+                found = true;
+                break;
+            }
+        }
+        assertTrue(found, "Expected contract address not found");
+    }
+
+/**
+ * @notice Test allowing an external contract to call the getHolographableContracts function.
+ * @dev This test verifies that an external contract can successfully call the getHolographableContracts function
+ * of the registry contract with the parameters 0 and 1.
+ * Refers to the hardhat test with the description 'Should allow external contract to call fn'
+ */
+    function testAllowExternalContractToCallGetHolographableContracts() public {
+        bytes memory encodeSignature = abi.encodeWithSignature('getHolographableContracts(uint256,uint256)', 0 , 1);
+        mockExternalCall.callExternalFn(address(registry), encodeSignature);
+        
+    }
+
+/*
+    getHolographableContractsLength
+*/
+
+/**
+ * @notice Test the validity of the length of holographable contracts returned by the registry.
+ * @dev This test verifies that the length of the list of holographable contracts returned by the registry
+ * matches the expected count of 5.
+ * Refers to the hardhat test with the description 'Should return valid _holographableContracts length'
+ */    
+    function testReturnValid_holographableContractsLength() public {
+        uint16 expectedHolographableContractsCount = 5;
+        uint256 length = registry.getHolographableContractsLength();
+        assertEq(length, expectedHolographableContractsCount);
+    }
+
+/**
+ * @notice Test allowing an external contract to call the getHolographableContractsLength function.
+ * @dev This test verifies that an external contract can successfully call the getHolographableContractsLength function
+ * of the registry contract.
+ * Refers to the hardhat test with the description 'Should allow external contract to call fn'
+ */
+    function testAllowExternalContractToCallGetHolographableContractsLength() public {
+        bytes memory encodeSignature = abi.encodeWithSignature('getHolographableContractsLength()');
+        mockExternalCall.callExternalFn(address(registry), encodeSignature);
+    }
+
+/*
+    isHolographedContract
+*/
+
+/**
+ * @notice Test the validation of a smart contract as a valid holographed contract.
+ * @dev This test checks if the smart contract at the address of sampleErc721Holographer
+ * is considered a valid holographed contract by the registry.
+ * It verifies that the function returns true for a valid holographed contract.
+ * Refers to the hardhat test with the description 'Should return true if smartContract is valid'
+ */
+    function testReturnTrueIfSmartContractIsValid() public {
+        bool isHolographed = registry.isHolographedContract(address(sampleErc721Holographer));
+        assertTrue(isHolographed);
```

Comment:
suggestion: you could save the variable and put the call of the function inside the assertTrue